ddcommon/src/cstr.rs

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+#[doc(hidden)]
+pub const fn validate_cstr_contents(bytes: &[u8]) {
+    if bytes[bytes.len() - 1usize] != b'\0' {
+        panic!("cstr must be null terminated");
+    }
+
+    let mut i = 0;
+    let check_len = if bytes[bytes.len() - 1] == b'\0' {
+        bytes.len() - 1usize
+    } else {
+        bytes.len()
+    };
+    while i < check_len {
+        if bytes[i] == b'\0' {
+            panic!("cstr string cannot contain null character outside of last element");
+        }
+        i += 1;
+    }
+}
+
+#[macro_export]
+macro_rules! cstr {
+    ($s:literal) => {{
+        let mut bytes = $s.as_bytes();
+        if bytes[bytes.len() - 1usize] != b'\0' {
+            bytes = concat!($s, "\0").as_bytes();
+        }
+
+        $crate::cstr::validate_cstr_contents(bytes);
+        unsafe { std::ffi::CStr::from_bytes_with_nul_unchecked(bytes) }
+    }};
+}
+
+#[macro_export]
+macro_rules! cstr_u8 {
+    ($s:literal) => {{
+        $crate::cstr::validate_cstr_contents($s);
+        unsafe { std::ffi::CStr::from_bytes_with_nul_unchecked($s as &[u8]) }
+    }};
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_cstr() {
+        assert_eq!(b"/dev/null", cstr!("/dev/null").to_bytes());
+        assert_eq!(b"/dev/null", cstr!("/dev/null\0").to_bytes());
+        assert_eq!(b"/dev/null", cstr_u8!(b"/dev/null\0").to_bytes());
```

Comment:
I don't know much about rust macros yet, can we combine `cstr!` and `cstr_u8` easily?